Kickstarter-iOS/Views/Controllers/ProjectPageViewController.swift

```diff
@@ -0,0 +1,415 @@
+      .observeForUI()
+      .observeValues { [weak self] project, _ in
+        /** FIXME: How we do this might change in https://kickstarter.atlassian.net/browse/NTV-195
+         self?.contentController?.configureWith(value: (project, refTag))
+         */
+        self?.shareViewModel.inputs.configureWith(shareContext: .project(project), shareContextView: nil)
+        self?.watchProjectViewModel.inputs
+          .configure(with: (project, KSRAnalytics.PageContext.projectPage, nil))
+      }
+
+    self.viewModel.outputs.configurePledgeCTAView
+      .observeForUI()
+      .observeValues { [weak self] value in
+        self?.pledgeCTAContainerView.configureWith(value: value)
+      }
+
+    self.viewModel.outputs.dismissManagePledgeAndShowMessageBannerWithMessage
+      .observeForControllerAction()
+      .observeValues { [weak self] message in
+        self?.dismiss(animated: true, completion: {
+          self?.messageBannerViewController?.showBanner(with: .success, message: message)
+        })
+      }
+
+    self.viewModel.outputs.popToRootViewController
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.navigationController?.popToRootViewController(animated: false)
+      }
+  }
+
+  private func bindSharingViewModel() {
+    self.shareViewModel.outputs.showShareSheet
+      .observeForControllerAction()
+      .observeValues { [weak self] controller, _ in self?.showShareSheet(controller) }
+  }
+
+  private func bindWatchViewModel() {
+    let button = self.navigationSaveButton.customView as? UIButton
+    button?.rac.accessibilityValue = self.watchProjectViewModel.outputs.saveButtonAccessibilityValue
+    button?.rac.selected = self.watchProjectViewModel.outputs.saveButtonSelected
+
+    self.watchProjectViewModel.outputs.generateImpactFeedback
+      .observeForUI()
+      .observeValues { generateImpactFeedback() }
+
+    self.watchProjectViewModel.outputs.generateNotificationSuccessFeedback
+      .observeForUI()
+      .observeValues { generateNotificationSuccessFeedback() }
+
+    self.watchProjectViewModel.outputs.generateSelectionFeedback
+      .observeForUI()
+      .observeValues { generateSelectionFeedback() }
+
+    self.watchProjectViewModel.outputs.showProjectSavedAlert
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.showProjectStarredPrompt()
+      }
+
+    self.watchProjectViewModel.outputs.goToLoginTout
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.goToLoginTout()
+      }
+
+    self.watchProjectViewModel.outputs.postNotificationWithProject
+      .observeForUI()
+      .observeValues { project in
+        NotificationCenter.default.post(
+          name: Notification.Name.ksr_projectSaved,
+          object: nil,
+          userInfo: ["project": project]
+        )
+      }
+  }
+
+  fileprivate func showProjectStarredPrompt() {
```

Comment:
should we move this to `SharedFunctions` since we are re-declaring this in two separate files?

Reply:
Wondering the same for the methods below as well.

Reply:
We will probably be deleting the old one after this is shipped though so let me know your thoughts.